### PR TITLE
Add workaround about using Stale Read with TiFlash

### PR DESCRIPTION
**Pull request by leiysky**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add workaround about using Stale Read with TiFlash.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/5875
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by Yisaer — APPROVED** [review]

File: as-of-timestamp.md [hunk 1]
```diff
@@ -10,6 +10,10 @@ summary: 了解如何使用 AS OF TIMESTAMP 语法读取历史数据。
 > **警告：**
 >
 > Stale Read 目前为实验特性。不推荐在生产环境下使用该特性。
+>
+> 需要注意的是，目前 Stale Read 特性无法和 TiFlash 一起使用。因此如果你的查询中带有 `AS OF TIMESTAMP` 并且可能从 TiFlash 副本读取数据，就有可能会遇到 `ERROR 1105 (HY000): stale requests require tikv backend` 这样的报错。
```
Comment (TomShawn) [inline as-of-timestamp.md:14]:
```suggestion
> 目前 Stale Read 特性无法和 TiFlash 一起使用。因此如果你的查询中带有 `AS OF TIMESTAMP` 并且 TiDB 可能从 TiFlash 副本读取数据，可能会遇到 `ERROR 1105 (HY000): stale requests require tikv backend` 错误。
```

File: as-of-timestamp.md [hunk 2]
```diff
@@ -10,6 +10,10 @@ summary: 了解如何使用 AS OF TIMESTAMP 语法读取历史数据。
...
+>
+> 你可以通过为使用 Stale Read 特性的查询禁用 TiFlash 副本来解决这个问题，推荐的方式有两种：一种是通过设置变量来禁用 TiFlash 副本 `set session tidb_isolation_read_engines='tidb,tikv'`；另一种是使用 [hint](https://docs.pingcap.com/zh/tidb/v5.1/optimizer-hints#read_from_storagetiflasht1_name--tl_name--tikvt2_name--tl_name-) 强制 TiDB 从 TiKV 读取数据。
```
Comment (TomShawn) [inline as-of-timestamp.md:16]:
```suggestion
> 要解决该问题，你需要为使用 Stale Read 特性的查询禁用 TiFlash 副本。要禁用 TiFlash 副本，你可以使用以下任一方法：

> + 通过设置变量来禁用 TiFlash 副本 `set session tidb_isolation_read_engines='tidb,tikv'。
> + 使用 [hint](https://docs.pingcap.com/zh/tidb/dev/optimizer-hints#read_from_storagetiflasht1_name--tl_name--tikvt2_name--tl_name-) 强制 TiDB 从 TiKV 读取数据。
```

**Comment by leiysky** [comment]
@TomShawn PTAL

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**